### PR TITLE
New resource: `azurerm_virtual_hub_security_partner_provider`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
API spec: https://docs.microsoft.com/en-us/rest/api/virtualnetwork/securitypartnerproviders/createorupdate

### CODE REVIEW

File: website/docs/r/security_partner_provider.html.markdown [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Security Partner Provider. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Security Partner Provider should exist. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Security Partner Provider should exist. Changing this forces a new resource to be created.
+
+* `security_provider_type` - (Required) The security provider name. Possible values are `ZScaler`, `IBoss` and `Checkpoint` is allowed. Changing this forces a new resource to be created.
```
Comment (katbyte) [inline website/docs/r/security_partner_provider.html.markdown:67]: Given the docs say name should this be
```suggestion
* `security_provider_name` - (Required) The security provider name. Possible values are `ZScaler`, `IBoss` and `Checkpoint` is allowed. Changing this forces a new resource to be created.
```

File: website/docs/r/security_partner_provider.html.markdown [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+* `virtual_hub_id` - (Optional) The ID of the Virtual Hub within which this Security Partner Provider should be created. Changing this forces a new resource to be created.
```
Comment (katbyte) [inline website/docs/r/security_partner_provider.html.markdown:69]: What happens when no vhub is specified?

File: azurerm/internal/services/network/tests/security_partner_provider_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+`, config)
+}
+
+func testAccAzureRMSecurityPartnerProvider_complete(data acceptance.TestData) string {
+	template := testAccAzureRMSecurityPartnerProvider_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_security_partner_provider" "test" {
+  name                   = "acctest-spp-%d"
```
Comment (katbyte) [inline azurerm/internal/services/network/tests/security_partner_provider_resource_test.go:216]: Could we ensure capitals work
```suggestion
  name                   = "acctest-SPP-%d"
```

File: azurerm/internal/services/network/tests/security_partner_provider_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  resource_group_name    = azurerm_resource_group.test.name
+  location               = azurerm_resource_group.test.location
+  virtual_hub_id         = azurerm_virtual_hub.test.id
+  security_provider_type = "ZScaler"
+
+  tags = {
+    ENV = "test"
```
Comment (katbyte) [inline azurerm/internal/services/network/tests/security_partner_provider_resource_test.go:223]: and here
```suggestion
    ENv = "Test"
```

File: azurerm/internal/services/network/registration.go [hunk 5]
```diff
@@ -102,6 +102,7 @@ func (r Registration) SupportedResources() map[string]*schema.Resource {
 		"azurerm_route_filter":                                                           resourceArmRouteFilter(),
 		"azurerm_route_table":                                                            resourceArmRouteTable(),
 		"azurerm_route":                                                                  resourceArmRoute(),
+		"azurerm_security_partner_provider":                                              resourceArmSecurityPartnerProvider(),
```
Comment (katbyte) [inline azurerm/internal/services/network/registration.go:105]: I think we should have this be a little more specific? what is this security partner applied to? vhubs? then should this be:
```suggestion
		"azurerm_virtual_hub_security_partner_provider":                                              resourceArmSecurityPartnerProvider(),
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the pr @neil-yechenwei  - overall looks good but i have some questions about resource and schema naming. please see comments left inline

File: website/docs/r/security_partner_provider.html.markdown [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+subcategory: "Network"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_security_partner_provider"
+description: |-
+  Manages a Security Partner Provider.
+---
+
+# azurerm_security_partner_provider
+
+Manages a Security Partner Provider.
```
Comment (katbyte) [inline website/docs/r/security_partner_provider.html.markdown:11]: Could we elaborate on what a SPP is and what it is applied to?

File: website/docs/r/security_partner_provider.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/security_partner_provider.html.markdown:67]: updated

File: website/docs/r/security_partner_provider.html.markdown (hunk 2 above)
Comment (neil-yechenwei) [reply website/docs/r/security_partner_provider.html.markdown:69]: The independent security partner provider would be created and it wouldn't be associated to virtual hub.

File: azurerm/internal/services/network/tests/security_partner_provider_resource_test.go (hunk 3 above)
Comment (neil-yechenwei) [reply azurerm/internal/services/network/tests/security_partner_provider_resource_test.go:216]: updated

File: azurerm/internal/services/network/tests/security_partner_provider_resource_test.go (hunk 4 above)
Comment (neil-yechenwei) [reply azurerm/internal/services/network/tests/security_partner_provider_resource_test.go:223]: updated

File: azurerm/internal/services/network/registration.go (hunk 5 above)
Comment (neil-yechenwei) [reply azurerm/internal/services/network/registration.go:105]: updated

File: website/docs/r/security_partner_provider.html.markdown (hunk 6 above)
Comment (neil-yechenwei) [reply website/docs/r/security_partner_provider.html.markdown:11]: updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated code. Please have a look. Thanks.

**Review by katbyte — APPROVED** [review]
Thanks @neil-yechenwei - LGTM 👍

**Pull request merged**